File: F/2269-find-the-k-beauty-of-a-number/main.go
package main

func divisorSubstrings(num int, k int) int {
	buf := []int{}
	n := num
	for n > 0 {
		buf = append(buf, n%10)
		n /= 10
	}

	v := 0
	factor := 1
	for i := 0; i < k; i++ {
		v += buf[i] * factor
		factor *= 10
	}
	factor /= 10

	ret := 0
	if v > 0 && num%v == 0 {
		ret++
	}

	for i := k; i < len(buf); i++ {
		v /= 10
		v = buf[i]*factor + v

		if v > 0 && num%v == 0 {
			ret++
		}
	}
	return ret
}

File: F/2269-find-the-k-beauty-of-a-number/main.rs
struct Solution {}

impl Solution {
  pub fn divisor_substrings(num: i32, k: i32) -> i32 {
    let mut base: i32 = 1;
    (0..k).for_each(|_| {
      base *= 10;
    });

    let mut n = num;
    let mut ans: i32 = 0;
    while n >= base / 10 {
      ans += if n % base != 0 && num % (n % base) == 0 {
        1
      } else {
        0
      };
      n /= 10;
    }
    ans
  }
}
